### ISSUE TITLE
Missing documentation on using touch functionalities in HA.

### ISSUE THREAD

**Issue by Wolveras** [home-assistant/home-assistant.github.io#19204]
### Feedback

I'm not sure if the current integration allows for it, but I am missing information on how to set up and leverage the touch functionality. This is a feature on the Nanoleaf Shapes and Canvas panels, and supported by Homekit for automation on that platform. See [this video](https://www.youtube.com/watch?v=w6gFLt9dlw4) for more information.

If there is no official support, are there any plans to do so? Are there any workarounds to let HA listen to the touch events? Adding this information to the documentation would make it more clear what the Nanoleaf integration can and cannot do at this point in time.

### URL

https://www.home-assistant.io/integrations/nanoleaf/

### Version

2021.9.2

**Comment by milanmeu** [comment]
Hi @Wolveras, thanks for your feedback.

**TL;DR**: _There is currently no support for touch, but it will be added in the future._

The integration hasn't gotten much love since it was added in 0.67 (2018).
I just started to make it comply with the new standards such as setup in the UI, new discovery setup and new internal light representation. These changes have been made and most are in 2021.9.

The next step is to clean up the integration and make it faster: Asynchronous and reduce the number of requests from 18 to 2 per minute. I hope to have these changes in 2021.10.

The new Nanoleaf Elements are not supported yet, support will be added for these lights first before adding new functionality.
There's still a long way to go before I can work on Touch Events, but it's on my list. https://github.com/users/milanmeu/projects/2#card-67372636.

I'm sorry, I can't indicate when Touch Events will be added. There is still so much to do and I am also working on other integrations. It will certainly be a few releases further.

> Are there any workarounds to let HA listen to the touch events?

I would say no because you can't easily poll these but you have to set up a stream. https://forum.nanoleaf.me/docs/openapi#_842h3097vbgq

> Adding this information to the documentation would make it more clear what the Nanoleaf integration can and cannot do at this point in time.

Sure, you can add this with the edit button at the bottom of the integration page.

I will try to keep you informed.

Have a nice day @Wolveras.

@milanmeu

**Comment by Wolveras** [comment]
> 
> 
> Hi @Wolveras, thanks for your feedback.
> 
> **TL;DR**: _There is currently no support for touch, but it will be added in the future._
> 
> The integration hasn't gotten much love since it was added in 0.67 (2018).
> I just started to make it comply with the new standards such as setup in the UI, new discovery setup and new internal light representation. These changes have been made and most are in 2021.9.
> 
> The next step is to clean up the integration and make it faster: Asynchronous and reduce the number of requests from 18 to 2 per minute. I hope to have these changes in 2021.10.
> 
> The new Nanoleaf Elements are not supported yet, support will be added for these lights first before adding new functionality.
> There's still a long way to go before I can work on Touch Events, but it's on my list. https://github.com/users/milanmeu/projects/2#card-67372636.
> 
> I'm sorry, I can't indicate when Touch Events will be added. There is still so much to do and I am also working on other integrations. It will certainly be a few releases further.
> 
> > Are there any workarounds to let HA listen to the touch events?
> 
> I would say no because you can't easily poll these but you have to set up a stream. https://forum.nanoleaf.me/docs/openapi#_842h3097vbgq
> 
> > Adding this information to the documentation would make it more clear what the Nanoleaf integration can and cannot do at this point in time.
> 
> Sure, you can add this with the edit button at the bottom of the integration page.
> 
> I will try to keep you informed.
> 
> Have a nice day @Wolveras.
> 
> @milanmeu

Hi @milanmeu,

Thank you for your swift and thorough reply.
It's good to hear that you're planning to extensively develop and expand the integration! I'll be looking forward to seeing them in HA in the future.
I've come to the same conclusion regarding workaround for the touch events - more trouble than it is worth at the moment (with my limited experience).

Thank you again, and have a nice day as well @milanmeu .
